demo_nodes_cpp/src/parameters/parameter_events_async.cpp

```diff
@@ -75,6 +77,7 @@ class ParameterEventsAsyncNode : public rclcpp::Node
   }
 
   // Set several different types of parameters.
+  DEMO_NODES_CPP_PUBLIC
   void queue_first_set_parameter_request()
```

Comment:
AFAICT, there's no reason for this method, or `queue_second_set_parameter_request`, to be public.
I would make these methods private and set the visibility to "hidden" with the macro `DEMO_NODES_CPP_LOCAL`.

I think the same can be done for all demo classes for methods that are not intended to be called externally.